pkg/config/setup.go

```diff
@@ -0,0 +1,169 @@
+package config
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/serum-errors/go-serum"
+
+	"github.com/warptools/warpforge/wfapi"
+)
+
+/*
+	The goal here is to reduce _undocumented global state_ and odd behavior handling it.
+	Things like env vars and pwd can change during runtime.
+	This is a highly probable source of undefined or confusing behavior.
+	It may even force strange behavior such as moving directories temporarily
+	before running an operation, which would preclude concurrent operations.
+	This may not be the _most correct_ or _best_ way to handle these issues.
+	But it is certainly a step forward.
+	This doesn't prevent changes that might be passed down to sub-procsses like rio, runc
+*/
+
+type State struct {
+	Env              map[string]string
```

Comment:
I think it would be more consistent to flatten the contents of this map into more fields rather than a map.

It would only be four more fields.

(Or if we end up deciding gathering these values into a struct doesn't provide significant value at all, and we just do package scope funcs for each thing: then disregard this comment; that approach probably naturally ends up with no env map either.)

Reply:
eh fine removed.